php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -1507,6 +1484,53 @@ public ExpressionTree completeConditionalExpr(ExpressionTree expression, Optiona
     return partial.isPresent() ? partial.get().complete(expression) : expression;
   }
 
+  public SeparatedListImpl<ExpressionTree> arguments(Optional<Tuple<ExpressionTree, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>>>> arguments) {
+    SeparatedListImpl<ExpressionTree> list;
+    if (arguments.isPresent()) {
+      list = separatedList(arguments.get().first(), arguments.get().second());
+    } else {
+      list = SeparatedListImpl.empty();
+    }
+
+    return list;
+  }
+
+  public AnonymousClassTree anonymousClass(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)